pkg/sql/alter_type.go

```diff
@@ -157,6 +157,44 @@ func (p *planner) performRenameTypeDesc(
 	return p.txn.Run(ctx, b)
 }
 
+func (p *planner) renameTypeValue(
+	params runParams, n *alterTypeNode, oldVal string, newVal string,
+) error {
+	newNameFound := false
+
+	enumMemberIndex := -1
+
+	// Do one pass to verify that the oldVal exists and there isn't already
+	// a member that is named newVal.
+	for i := range n.desc.EnumMembers {
+		member := n.desc.EnumMembers[i]
+		if member.LogicalRepresentation == oldVal {
+			enumMemberIndex = i
+		} else if member.LogicalRepresentation == newVal {
+			newNameFound = true
```

Comment:
Why do you need to record this for later? Can't you just error out here already

Reply:
Updated